collect_app/src/test/java/org/odk/collect/android/entities/EntitiesRepositoryTest.kt

```diff
@@ -215,13 +216,54 @@ abstract class EntitiesRepositoryTest {
     fun `#save can save multiple entities`() {
         val repository = buildSubject()
 
-        val wine = Entity("wines", "1", "Léoville Barton 2008")
-        val whisky = Entity("whiskys", "2", "Lagavulin 16")
+        val wine = Entity.New("wines", "1", "Léoville Barton 2008")
+        val whisky = Entity.New("whiskys", "2", "Lagavulin 16")
         repository.save(wine, whisky)
 
         assertThat(repository.getLists(), containsInAnyOrder("wines", "whiskys"))
     }
 
+    @Test
+    fun `#save assigns an index to each entity in insert order when saving multiple entities`() {
```

Comment:
They don't strictly need to be in insert order most likely, but this behaviour is much easier to test, and will mean the resulting multiplicity within form entry will match the order in the CSVs.